design-documents/service-isolation/cicd-pipeline.md

```diff
@@ -0,0 +1,48 @@
+# CI/CD pipeline for service isolation
+
+## About
+
+In the scope of service isolation we want to have CI/CD process that would allow
+* Running tests on change set and update preview environment
+* Automate PR delivery to mainline
+
+And in addition to that support the following scenarios that can be used internally and by system integrators
+* Update stage environment
+* Update production environment
+* Support blue/green deployment
+
+Steps that are currently not automated
+* Creating builds for change set
+* Updating preview environment (teams use for testing)
+* Updating stage environment (teams use for demo purposes)
+
+Pipelines should be configured by code and/or convention. Triggered by pushing code. There are should be minimal/or no interaction with UI (with the exception of PR approvals) if the pipeline is successful.
+
+## Design
+
+Branches named by convention. Branches that start with feature contain clean feature related changes. Pushing changes to feature branches triggers CI pipeline below. User receive notification via slack/email with links to builds.
+
+![New CI pipeline](img/new-ci-pipeline.png)
```

Comment:
I would expect the GitHub Pull Request to be the first thing in this chart that triggers the tests and (if passed) the preview environment, all in one shot.

If a PR is not ready to be merged in, we can also use [draft pull requests](https://github.blog/2019-02-14-introducing-draft-pull-requests/).